presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -0,0 +1,182 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.esri.core.geometry.Geometry;
+import com.esri.core.geometry.GeometryEngine;
+import com.esri.core.geometry.OperatorImportFromESRIShape;
+import com.esri.core.geometry.SpatialReference;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import static java.util.Objects.requireNonNull;
+
+public final class GeometryUtils
+{
+    public enum OGCType
+    {
+        UNKNOWN(0),
+        ST_POINT(1),
+        ST_LINESTRING(2),
+        ST_POLYGON(3),
+        ST_MULTIPOINT(4),
+        ST_MULTILINESTRING(5),
+        ST_MULTIPOLYGON(6);
+
+        private final int index;
+
+        private OGCType(int index)
+        {
+            this.index = index;
+        }
+
+        public int getIndex()
+        {
+            return this.index;
+        }
+    }
+
+    public static final String POINT = "Point";
+    public static final String LINE_STRING = "LineString";
+    public static final String POLYGON = "Polygon";
+    public static final String MULTI_POINT = "MultiPoint";
+    public static final String MULTI_LINE_STRING = "MultiLineString";
+    public static final String MULTI_POLYGON = "MultiPolygon";
+    public static final int WKID_UNKNOWN = 0;
+
+    private static final int SIZE_WKID = 4;
+    private static final int SIZE_TYPE = 1;
+    private static final OGCType[] values = OGCType.values();
+
+    private GeometryUtils() {}
+
+    public static OGCGeometry deserialize(Slice shape)
+    {
+        requireNonNull(shape, "shape is null");
+        int wkid = shape.getInt(0);
+        ByteBuffer shapeBuffer = getShapeByteBuffer(shape);
+        if (shapeBuffer.limit() < 4) {
+            return null;
+        }
+        else {
+            if (shapeBuffer.getInt(0) == Geometry.Type.Unknown.value()) {
+                return null;
+            }
+            else {
+                SpatialReference spatialReference = null;
+                if (wkid != WKID_UNKNOWN) {
+                    spatialReference = SpatialReference.create(wkid);
+                }
+                Geometry esriGeom = OperatorImportFromESRIShape.local().execute(0, Geometry.Type.Unknown, shapeBuffer);
+                OGCGeometry createdGeom = OGCGeometry.createFromEsriGeometry(esriGeom, spatialReference);
+                return createdGeom;
+            }
+        }
+    }
+
+    public static final OGCType getOGCType(int value)
+    {
+        if (value < 0 || value >= values.length) {
+            return OGCType.UNKNOWN;
+        }
+        else {
+            return values[value];
+        }
+    }
+
+    public static Slice serialize(OGCGeometry ogcGeometry)
+    {
+        int wkid;
+        try {
+            wkid = ogcGeometry.SRID();
+        }
+        catch (NullPointerException npe) {
```

Comment:
What's the rational for catching NPE here and in line 139? Is this necessary to allow calling serialize with a null argument? If so, it would be cleaner to check ogcGeometry for null in the very beginning of this method.